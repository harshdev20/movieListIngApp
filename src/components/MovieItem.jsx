import React from "react";
import LazyImage from "./LazyImage.jsx";

function MovieItem({ movie }) {
  let srcImage = movie["poster-image"].includes("posterthatismissing")
    ? `/slices/placeholder_for_missing_posters.png`
    : `/slices/${movie["poster-image"]}`;

  let name =
    movie.name.length > 15 ? movie.name.slice(0, 10) + "..." : movie.name;
  return (
    <div className="lg:mb-16 pr-5 mb-4">
      <LazyImage src={process.env.PUBLIC_URL + srcImage} />
      <p className="text-gray-500 lg:text-sm text-xs lg:my-2 ">{name} </p>
    </div>
  );
}

export default MovieItem;
