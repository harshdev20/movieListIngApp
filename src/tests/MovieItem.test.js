import React from "react";
import { create, act } from "react-test-renderer";
import MovieItem from "../components/MovieItem";

it("should snapshot test movieItem", () => {
  let root;
  act(() => {
    root = create(
      <MovieItem
        movie={{
          name: "The Birds",
          "poster-image": "poster1.jpg",
        }}
      />
    );
  });

  expect(root.toJSON()).toMatchSnapshot();
});
