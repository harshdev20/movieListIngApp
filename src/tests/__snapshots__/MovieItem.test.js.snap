// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot test movieItem 1`] = `
<div
  className="lg:mb-16 pr-5 mb-4"
>
  <span
    className=" lazy-load-image-background blur"
    style={
      Object {
        "backgroundImage": "",
        "backgroundSize": "",
        "color": "transparent",
        "display": "inline-block",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span
      className=" min-w-full lg:w-48"
      style={
        Object {
          "display": "inline-block",
        }
      }
    />
  </span>
  <p
    className="text-gray-500 lg:text-sm text-xs lg:my-2 "
  >
    The Birds
     
  </p>
</div>
`;
